Celeste.Mod.mm/Patches/CrystalStaticSpinner.cs

```diff
@@ -39,6 +78,85 @@ public patch_CrystalStaticSpinner(Vector2 position, bool attachToSolid, CrystalC
             orig_Awake(scene);
         }
 
+        public extern void orig_CreateSprites();
+        public void CreateSprites()
+        {              
+            if (!this.expanded)
+            {
+                Calc.PushRandom(this.randomSeed);
+                List<MTexture> atlasSubtextures;
+                if (this.directory == "")
+                {
+                    atlasSubtextures = GFX.Game.GetAtlasSubtextures(patch_CrystalStaticSpinner.fgTextureLookup[this.color]);
+                } else
+                {
+                    atlasSubtextures = GFX.Game.GetAtlasSubtextures(this.directory + "/fg");
+                }    
+                MTexture mtexture = Calc.Random.Choose(atlasSubtextures);
+                Image image;
+                if (!this.SolidCheck(new Vector2(base.X - 4f, base.Y - 4f)))
+                {
+                    image = new Image(mtexture.GetSubtexture(0, 0, 14, 14, null)).SetOrigin(12f, 12f);
+                    image.Color = Calc.HexToColor(this.tint);
+                    base.Add(image);
+                }
+                if (!this.SolidCheck(new Vector2(base.X + 4f, base.Y - 4f)))
+                {
+                    image = new Image(mtexture.GetSubtexture(10, 0, 14, 14, null)).SetOrigin(2f, 12f);
+                    image.Color = Calc.HexToColor(this.tint);
+                    base.Add(image);
+                }
+                if (!this.SolidCheck(new Vector2(base.X + 4f, base.Y + 4f)))
+                {
+                    image = new Image(mtexture.GetSubtexture(10, 10, 14, 14, null)).SetOrigin(2f, 2f);
+                    image.Color = Calc.HexToColor(this.tint);
+                    base.Add(image);
+                }
+                if (!this.SolidCheck(new Vector2(base.X - 4f, base.Y + 4f)))
+                {
+                    image = new Image(mtexture.GetSubtexture(0, 10, 14, 14, null)).SetOrigin(12f, 2f);
+                    image.Color = Calc.HexToColor(this.tint);
+                    base.Add(image);
+                }
+                foreach (Entity entity in base.Scene.Tracker.GetEntities<CrystalStaticSpinner>())
+                {
+                    CrystalStaticSpinner crystalStaticSpinner = (CrystalStaticSpinner)entity;
+                    if (crystalStaticSpinner != this && crystalStaticSpinner.AttachToSolid == this.AttachToSolid && crystalStaticSpinner.X >= base.X && (crystalStaticSpinner.Position - this.Position).Length() < 24f)
+                    {
+                        this.AddSprite((this.Position + crystalStaticSpinner.Position) / 2f - this.Position);
+                    }
+                }
+                base.Scene.Add(this.border = new patch_CrystalStaticSpinner.Border(this, this.filler));
+                this.expanded = true;
+                Calc.PopRandom();
+            }
+        }
+
+        public extern void orig_AddSprite(Vector2 offset);
```

Comment:
Same as `CreateSprites`: the `orig_` variant is unused, and the method is being redefined just to change the texture lookup.